backend/dataall/base/aws/cognito.py

```diff
@@ -34,11 +34,33 @@ def get_user_emailids_from_group(self, groupName):
             if envname in ['local', 'dkrcompose']:
                 log.error('Local development environment does not support Cognito')
                 return ['anonymous@amazon.com']
-            log.error(f'Failed to get email ids for Cognito group {groupName} due to {e}')
+            log.error(f'Failed to get usernames for Cognito group {groupName} due to {e}')
             raise e
         else:
             return group_email_ids
 
+    def get_user_list_from_group(self, groupName):
+        try:
+            envname = os.getenv('envname', 'local')
+            parameter_path = f'/dataall/{envname}/cognito/userpool'
+            ssm = boto3.client('ssm', region_name=os.getenv('AWS_REGION', 'eu-west-1'))
+            user_pool_id = ssm.get_parameter(Name=parameter_path)['Parameter']['Value']
+            paginator = self.client.get_paginator('list_users_in_group')
+            pages = paginator.paginate(UserPoolId=user_pool_id, GroupName=groupName)
+            cognito_user_list = []
+            for page in pages:
+                cognito_user_list += page['Users']
+            group_usernames = [user['Username'] for user in cognito_user_list]
+        except Exception as e:
+            envname = os.getenv('envname', 'local')
+            if envname in ['local', 'dkrcompose']:
+                log.error('Local development environment does not support Cognito')
+                return ['anonymous@amazon.com']
+            log.error(f'Failed to get email ids for Cognito group {groupName} due to {e}')
```

Comment:
nit: should this be `usernames` not `email ids`

Reply:
Thanks, I changed the wrong error message.  Fixed.